data-prepper-plugins/encryption-plugin/src/main/java/org/opensearch/dataprepper/plugins/encryption/AwsContext.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.encryption;
+
+import org.opensearch.dataprepper.aws.api.AwsCredentialsConfig;
+import org.opensearch.dataprepper.aws.api.AwsCredentialsOptions;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+
+import java.util.UUID;
+
+public class AwsContext {
+    private final AwsCredentialsConfig awsCredentialsConfig;
+
+    public AwsContext(final AwsCredentialsConfig awsCredentialsConfig) {
+        this.awsCredentialsConfig = awsCredentialsConfig;
+    }
+
+    public AwsCredentialsProvider getOrDefault() {
```

Comment:
This code looks similar to what we already support in the aws-plugin.

Reply:
This class was refactored out from `kafka-plugins` and I did not see similar code in `aws-plugin`. The plan is to replace those classes in `kafka-plugins` with the counterparts here.

Or do you think it is better to refactor it into `aws-plugin` and let `encryption-plugin` depend on `aws-plugin`?

Reply:
Similar methods are provided in [AwsCredentialsSupplier](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/aws-plugin-api/src/main/java/org/opensearch/dataprepper/aws/api/AwsCredentialsSupplier.java) interface, and detailed actions through [CredentialsProviderFactory](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/CredentialsProviderFactory.java). 

Kafka plugins also use it in [AwsContext](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/common/aws/AwsContext.java)

Reply:
I refactored AwsContext into aws-plugin-api so that it will be reused by other plugins

Reply:
@dlvenable Can you also take a look? I'm good with other changes. Approving.